app/lib/account_deletion_config.rb

```diff
@@ -1,27 +1,38 @@
 # frozen_string_literal: true
 
-module AccountDeletionConfig
-  module_function
+class AccountDeletionConfig
+  def self.configure(config = {})
+    @configuration = new(config)
+  end
+
+  def self.config
+    (@configuration || configure).config
```

Comment:
or maybe even better with: (since you're doing the same twice).

```
def self.configuration
  @configuration || configure
end
```

Reply:
mmmm it is not really the same... 🤔 `.config` means the `@config` value inside the instance.

Reply:
Ok, explanation by Michal:

```ruby
def self.configuration
  @configuration || configure
end

def self.config
    configuration.config
end
```

what’s the difference between this and this:
```ruby
def self.config
    (@configuration || configure).config
end
```